src/hooks/use-newsletter-subscription.js

```diff
@@ -0,0 +1,94 @@
+import { useEnv } from "../context/env.context";
+import useProfileApi from "./use-profile-api";
+import { useAuth0 } from "@auth0/auth0-react";
+import axios from "axios";
+import { useState, useEffect, useCallback } from "react";
+
+export default function useNewsLetterSubscriptionAPI(){
+
+  const { apiServerUrl } = useEnv();
+  const { fetchUser } = useProfileApi();
+  const { getAccessTokenSilently, user } = useAuth0();
+
+  
+
+  const makeRequest = useCallback(async (options) => {
+    
+      try {
+        
+          if (options.authenticated) {
+              const token = await getAccessTokenSilently();
+             
+              options.config.headers = {
+                  ...options.config.headers,
+                  Authorization: `Bearer ${token}`,
+              };
+          }
+          const return_val = await axios(options.config)
+          const { data } = return_val
+          return {data}
+
+      } catch (error) {
+
+          if (axios.isAxiosError(error) && error.response) {
+              return error.response;
+          }
+
+          return error.message;
+      }
+  }, [getAccessTokenSilently]);
+
+  /**
+   * calls either the unsubscribe or subscribe based on the param
+   * @param {subscribe, user_id} strings
+   */
+  async function subscribe(subscription_state, user_id){
```

Comment:
It is a string of either `subscribe` or `unsubscribe` based on onClick event emitter from a button on either the subscribe or unsubscribe page which defines the path of the API call. 
eg api/newsletter-subs/unsubscribe/${user_id} or api/newsletter-subs/subscribe/${user_id}

This would mean that I wouldn't have to write the same function twice.

Is this an acceptable convention? @gregv